### PR TITLE
WIP: add discrete_events to DSL

### PR DESCRIPTION
**Pull request by tstrey**
Adds `@discrete_events` option to the DSL such that reaction_system can be can be written in the following way:

```
rn = @reaction_network rxs begin
     @variables t
     @parameters k
     @species A(t) B(t)
     @discrete_events begin
          (t == 2.0) => [B ~ 0.0]
     end
     k, A --> B
end
```

### CODE REVIEW

File: src/reaction_network.jl [hunk 1]
```diff
@@ -123,7 +123,11 @@ const forbidden_variables_error = let
 end
 
 # Declares the keys used for various options.
+<<<<<<< Updated upstream
 const option_keys = (:species, :parameters, :variables, :ivs)
+=======
+const option_keys = (:species, :parameters, :variables, :ivs, :discrete_events)
+>>>>>>> Stashed changes
```
Comment (isaacsas) [inline src/reaction_network.jl:131]: This looks like something has gotten messed up when trying to merge master into your PR. There should be such comments listed.

**Review by isaacsas — COMMENTED** [review]
If you think the macro is working you should add some tests that use it and make sure the correct system is generated (for example, generate a system with the macro and by hand symbolically and test they give similar results when simulated).

**Comment by tstrey** [comment]
This works for systems like 

```
rn = @reaction_network a begin
              @parameters k
              @variables V(t)
              @species B(t) C(t)
              @discrete_events begin
                      a == 0 => [a ~ a+1, b ~ b-1]
               end
               k, C --> B
        end
```

**Comment by tstrey** [comment]
Needs to be modified for multiple discrete_events.

**Comment by isaacsas** [comment]
@TorkelE ended up adding this as part of Catalyst 14, but thanks for your getting this started!

**Pull request closed without merging**